lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx

```diff
@@ -4,167 +4,92 @@
-    return shallow(<NxCheckbox { ...simpleProps } { ...additionalProps } />);
-  }
-
-  it('renders a <label> containing a checkbox <input> and visible box', function() {
-    const shallowRender = getShallowComponent();
-
-    expect(shallowRender).toMatchSelector('label.nx-checkbox');
-
-    expect(shallowRender.find('input')).toHaveProp('type', 'checkbox');
-    expect(shallowRender.find('input')).toHaveProp('id', 'checkbox-id');
-    expect(shallowRender.find('input')).toHaveClassName('nx-checkbox__input');
-
-    expect(shallowRender.find('.nx-checkbox__box')).toExist();
-
-    // space is nbsp for empty box
-    expect(shallowRender).toHaveText(' ');
+  const minimalProps: Props & RefAttributes<HTMLLabelElement> = {
+        isChecked: false
+      },
+      quickRender = rtlRender(NxCheckbox, minimalProps),
+      renderEl = rtlRenderElement(NxCheckbox, minimalProps);
+
+  it('renders a <label> containing a checkbox role <input>', function() {
+    const component = renderEl()!,
+        input = within(component).getByRole('checkbox');
+
+    expect(component.tagName).toBe('LABEL');
+    expect(component).toContainElement(input);
   });
 
-  it('adds classes specified with the className prop', function() {
-    const component = getShallowComponent({ className: 'foo' });
+  it('adds specified classNames to the top-level element in addition to the defaults', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-checkbox');
-  });
-
-  it('renders a faCheck FontAwesomeIcon inside the box when checked', function() {
-    expect(getShallowComponent()).not.toContainReact(<FontAwesomeIcon icon={faCheck} />);
+    expect(el).toHaveClass('foo');
 
-    // icon and no nbsp when checked
-    expect(getShallowComponent({ isChecked: true }).find('.nx-checkbox__box'))
-        .toMatchElement(<span className="nx-checkbox__box"><FontAwesomeIcon icon={faCheck} /></span>);
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('renders no label text if none is provided', function() {
-    // space is nbsp for empty box
-    expect(shallow(<NxCheckbox checkboxId="checkbox-id" isChecked={false} onChange={() => {}} />)).toHaveText(' ');
-  });
+  it('passes additional attrs to the top-level element', function() {
+    const el = renderEl({ id: 'foo', lang: 'en-US' });
 
-  it('adds the nx-radio-checkbox--disabled class if disabled is set', function() {
-    expect(getShallowComponent()).not.toHaveClassName('nx-radio-checkbox--disabled');
-    expect(getShallowComponent({ disabled: true })).toHaveClassName('nx-radio-checkbox--disabled');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en-US');
   });
 
-  it('adds the tm-checked class if isChecked is true, and the tm-unchecked class if it is false', function() {
```

Comment:
This test should be kept/updated.  These classes are for external user analytics needs

Reply:
Updated here 784425a7d80cafc654eadcb4bbf42a4371e06189, thanks.